library/std/src/sync/nonpoison/mutex.rs

```diff
@@ -0,0 +1,616 @@
+#[cfg(all(test, not(any(target_os = "emscripten", target_os = "wasi"))))]
+mod tests;
+
+use crate::cell::UnsafeCell;
+use crate::fmt;
+use crate::marker::PhantomData;
+use crate::mem::ManuallyDrop;
+use crate::ops::{Deref, DerefMut};
+use crate::ptr::NonNull;
+use crate::sys::sync as sys;
+
+/// A mutual exclusion primitive useful for protecting shared data
```

Comment:
The docs for this one could probably just have a single example and then suggest looking at `std::sync::poison::Mutex` for more, so that we don't have the detailed documentation repeated in multiple locations. Also make sure the doctests are using the right type here.